crates/fuel-core/src/query/message.rs

```diff
@@ -337,3 +341,184 @@ pub fn message_status(
+            TxId,
+        },
+        fuel_types::BlockHeight,
+        services::txpool::TransactionStatus,
+        tai64::Tai64,
+    };
+
+    use super::{
+        message_receipts_proof,
+        MessageProofData,
+    };
+
+    pub struct FakeDB {
+        pub blocks: HashMap<BlockHeight, CompressedBlock>,
+        pub transaction_statuses: HashMap<TxId, TransactionStatus>,
+        pub receipts: HashMap<TxId, Vec<Receipt>>,
+    }
+
+    impl FakeDB {
+        fn new() -> Self {
+            Self {
+                blocks: HashMap::new(),
+                transaction_statuses: HashMap::new(),
+                receipts: HashMap::new(),
+            }
+        }
+
+        fn insert_block(&mut self, block_height: BlockHeight, block: CompressedBlock) {
+            self.blocks.insert(block_height, block);
+        }
+
+        fn insert_transaction_status(
+            &mut self,
+            transaction_id: TxId,
+            status: TransactionStatus,
+        ) {
+            self.transaction_statuses.insert(transaction_id, status);
+        }
+
+        fn insert_receipts(&mut self, transaction_id: TxId, receipts: Vec<Receipt>) {
+            self.receipts.insert(transaction_id, receipts);
+        }
+    }
+
+    impl MessageProofData for FakeDB {
+        fn block(&self, id: &BlockHeight) -> fuel_core_storage::Result<CompressedBlock> {
+            self.blocks.get(id).cloned().ok_or(not_found!("Block"))
+        }
+
+        fn transaction_status(
+            &self,
+            transaction_id: &TxId,
+        ) -> fuel_core_storage::Result<TransactionStatus> {
+            self.transaction_statuses
+                .get(transaction_id)
+                .cloned()
+                .ok_or(not_found!("Transaction status"))
+        }
+
+        fn block_history_proof(
+            &self,
+            _message_block_height: &BlockHeight,
+            _commit_block_height: &BlockHeight,
+        ) -> fuel_core_storage::Result<MerkleProof> {
+            // Unused in current tests
+            Ok(MerkleProof::default())
+        }
+
+        fn receipts(
+            &self,
+            transaction_id: &TxId,
+        ) -> fuel_core_storage::Result<Vec<Receipt>> {
+            self.receipts
+                .get(transaction_id)
+                .cloned()
+                .ok_or(not_found!("Receipts"))
+        }
+    }
+
+    // Test will try to get the message receipt proof with a block with only valid transactions
+    // Then add an invalid transaction and check if the proof is still the same (meaning the invalid transaction was ignored)
+    #[test]
+    fn test_message_receipts_proof_ignore_failed() {
+        // Create a fake database
+        let mut database = FakeDB::new();
+
+        // Given
+        // Create a block with a valid transaction and receipts
+        let mut block = CompressedBlock::default();
+        let valid_tx_id = Bytes32::new([1; 32]);
+        let mut valid_tx_receipts = vec![];
+        for i in 0..100 {
```

Comment:
Why do we need `100` receipts?=D

Reply:
Its in order to build a bigger tree and have some path for the proof. It could be 10 also.